packages/gatsby-plugin-fastify/src/gatsby/serverRoutes.ts

```diff
@@ -9,9 +9,9 @@ export async function getServerSideRoutes(pageData: PluginData) {
 
   for (const page of pages.values()) {
     if (page?.mode === "DSG" || page?.mode === "SSR") {
-      const { path, mode } = page;
+      const { path, mode, matchPath } = page;
       routes.push({
-        path,
+        path: matchPath || path,
```

Comment:
There's a difference between fastify path matching and what Gatsby expects. The client route handler has a string modifier to fix this. Let's move that into a utility function and use that here as well. Let me know if you have questions.

Reply:
This also might better be `??`....not 100% though.

Reply:
Agree on the matchPath handling. Will fix

In the case of the matchPath being an empty string, then it would make a difference I guess. That will most likely never happen. But if we ask ourselves what is the most correct operator. Then the `??` is the answer Will fix